src/lib/adapters/WebDav.ts

```diff
@@ -100,8 +100,12 @@ export default class WebDavAdapter extends CachingAdapter {
       res = await this.checkLock()
       if (res.status === 200) {
         if (res.headers['Last-Modified']) {
-          const date = new Date(res.headers['Last-Modified'])
-          startDate = date.valueOf()
+          const lastModifiedMs = new Date(res.headers['Last-Modified']).valueOf()
+          // if lastModifiedMs is greater than startDate, the lock file has been updated by someone else, reset the timeout logic
+          if (lastModifiedMs > startDate) {
+            startDate = lastModifiedMs
+            i = 0;
+          }
```

Comment:
So your intention is to avoid resetting startDate when lastModified is earlier than the current startDate? That would defeat the purpose of this mechanism I believe, as if the LastModified value is way earlier and becomes the new startDate, the loop stops, because the elapsed time since startDate is larger than maxTimeout and since res.status is 200 floccus should end up ignoring the lock and continue with sync. Am I missing something?